src/file/format/dhall.rs

```diff
@@ -0,0 +1,55 @@
+use crate::map::Map;
+use std::error::Error;
+
+use crate::{
+    error::Unexpected,
+    value::{Value, ValueKind},
+    ConfigError,
+};
+
+pub fn parse(
+    uri: Option<&String>,
+    text: &str,
+) -> Result<Map<String, Value>, Box<dyn Error + Send + Sync>> {
+    let value = from_dhall_value(uri, serde_dhall::from_str(text).parse()?);
+    match value.kind {
+        ValueKind::Table(map) => Ok(map),
+        ValueKind::Nil => Err(Unexpected::Unit),
+        ValueKind::Boolean(value) => Err(Unexpected::Bool(value)),
+        ValueKind::Integer(value) => Err(Unexpected::Integer(value)),
+        ValueKind::Float(value) => Err(Unexpected::Float(value)),
+        ValueKind::String(value) => Err(Unexpected::Str(value)),
+        ValueKind::Array(value) => Err(Unexpected::Seq),
+    }
+    .map_err(|err| ConfigError::invalid_root(uri, err))
+    .map_err(|err| Box::new(err) as Box<dyn Error + Send + Sync>)
```

Comment:
Only `json5` seems to have logic similar to this, and I noticed the [original PR references adopting it](https://github.com/mehcode/config-rs/pull/218#discussion_r680268947).

https://github.com/mehcode/config-rs/blob/d7c165602199d045cdee3aab23ff69d7269de7e3/src/file/format/json5.rs#L23-L36

This seems misunderstood as the method used to get the value differs? When the value returned is `ValueKind` like the other parsers, from what I understand this should not be necessary? I could be mistaken though 🤷‍♂️